releasewarrior/helpers.py

```diff
@@ -186,6 +188,41 @@ def validate_data_repo_updated(logger, config):
 
     return True
 
+
+def validate_rw_repo(logger, config):
+    ### data repo state file
+    state_file = os.path.join(config['releasewarrior_data_repo'], 'state.yml')
+    min_sha = None
+    if os.path.isfile(state_file):
+        with open(state_file) as fh:
+            state = yaml.load(fh)
+        min_sha = state.get('min-rw-sha')
+    else:
+        logger.warning('no release data state file')
+        return
+
+    if not min_sha:
+        logger.warning('no release data information on minimum sha for releasewarrior')
+
+    if not re.match('[0-9a-fA-F]', min_sha):
+        logger.fatal('min sha ({}) is invalid format'.format(min_sha))
+        sys.exit(1)
+
+    repo = Repo(RW_REPO)
+    if repo.is_dirty():
+        logger.warning("releasewarrior repo dirty")
+    upstream = find_upstream_repo(
+        repo, logger, config, pattern_key='upstream_rw_repo_url_pattern',
+        simplified_pattern_key='simplified_rw_repo_url')
+
+    logger.info("ensuring releasewarrior repo is newer than {} from {}".format(min_sha, upstream))
+    try:
+        repo.git.merge_base('--is-ancestor', min_sha, repo.head.commit)
```

Comment:
ah, I though this tried to pull, merge, and commit. iiuc now, this merely tries to see if local rw repo has latest remote csets